### PR TITLE
feat(one-app): removed tenancy or tenant instances

### PR DESCRIPTION
**Pull request by NarmeenNaveedAhmed**
## Description
This PR is for removing/replacing(where needed) all the instances of tenancy and tenant.

## Motivation and Context
This is to ensure consistency in the code base and to keep it generic.

## How Has This Been Tested?
-

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality) 
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?
The idea is to keep it as generic so specific keywords like tenant or tenancy do not cause any confusion for developers.

### CODE REVIEW

File: __tests__/integration/one-app.spec.js [hunk 1]
```diff
@@ -638,7 +638,7 @@ describe('Tests that require Docker setup', () => {
         await expect(searchForRequerstLog).resolves.toMatch(requestLogRegex);
       });
 
-      it('log gets updated when Tenancy Root module gets updated', async () => {
+      it('log gets updated when Application Root module gets updated', async () => {
```
Comment (nellyk) [inline __tests__/integration/one-app.spec.js]: what do you think about changing this to only _Root module_  we might not need the Application part

File: prod-sample/sample-modules/frank-lloyd-root/0.0.0/README.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # frank-lloyd-root
 
-Tenancy root module for integration tests. This specific version has a valid configuration.
\ No newline at end of file
+Application root module for integration tests. This specific version has a valid configuration.
```
Comment (nellyk) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.0/README.md]: same case here,having it as  root module might still make work

File: __tests__/integration/one-app.spec.js (hunk 1 above)
Comment (NarmeenNaveedAhmed) [reply __tests__/integration/one-app.spec.js]: sounds good!

File: __tests__/integration/one-app.spec.js [hunk 3]
```diff
@@ -515,7 +515,7 @@ describe('Tests that require Docker setup', () => {
               expect(JSON.parse(needyFrankModuleState)).toMatchSnapshot();
             });
 
-            describe('uses tenant provided fetch', () => {
+            describe('uses application provided fetch', () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
            describe('uses root module provided fetch', () => {
```

File: __tests__/integration/one-app.spec.js [hunk 4]
```diff
@@ -106,7 +106,7 @@ describe('Tests that require Docker setup', () => {
       expect(response.headers._headers).not.toHaveProperty('access-control-allow-credentials');
     });
 
-    describe('tenant without corsOrigins set', () => {
+    describe('Application without corsOrigins set', () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
    describe('root module appConfig without corsOrigins set', () => {
```

File: __tests__/integration/one-app.spec.js [hunk 5]
```diff
@@ -274,7 +274,7 @@ describe('Tests that require Docker setup', () => {
         });
       });
 
-      describe('tenant module config', () => {
+      describe('Application module config', () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
      describe('root module appConfig', () => {
```

File: __tests__/integration/one-app.spec.js [hunk 6]
```diff
@@ -571,8 +571,8 @@ describe('Tests that require Docker setup', () => {
       });
     });
 
-    describe('module requires SafeRequest Restricted Attributes not provided by tenant module', () => {
-      const requestRestrictedAttributesRegex = /Error: Tenant root must extendSafeRequestRestrictedAttributes with cookies: \[macadamia,homebaked\]/;
+    describe('module requires SafeRequest Restricted Attributes not provided by the module', () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
    describe('module requires SafeRequest Restricted Attributes not provided by the root module', () => {
```

File: __tests__/integration/one-app.spec.js [hunk 7]
```diff
@@ -571,8 +571,8 @@ describe('Tests that require Docker setup', () => {
...
+      const requestRestrictedAttributesRegex = /Error: Application root must extendSafeRequestRestrictedAttributes with cookies: \[macadamia,homebaked\]/;
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
      const requestRestrictedAttributesRegex = /Error: Root module must extendSafeRequestRestrictedAttributes with cookies: \[macadamia,homebaked\]/;
```

File: __tests__/integration/one-app.spec.js [hunk 8]
```diff
@@ -606,7 +606,7 @@ describe('Tests that require Docker setup', () => {
       });
     });
 
-    test('app calls loadModuleData to run async requests using Tenant provided fetchClient', async () => {
+    test('app calls loadModuleData to run async requests using application provided fetchClient', async () => {
```
Comment (JAdshead) [inline __tests__/integration/one-app.spec.js]:
```suggestion
    test('app calls loadModuleData to run async requests using root module provided fetchClient', async () => {
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 9]
```diff
@@ -269,7 +269,7 @@ describe('onModuleLoad', () => {
     expect(getModulesUsingExternals()).toEqual([]);
   });
 
-  it('validates csp added to tenant module', () => {
+  it('validates csp added to the module', () => {
```
Comment (JAdshead) [inline __tests__/server/utils/onModuleLoad.spec.jsx]:
```suggestion
  it('validates csp added to the root module', () => {
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 10]
```diff
@@ -278,7 +278,7 @@ describe('onModuleLoad', () => {
     expect(callOnModuleLoad).toThrowErrorMatchingSnapshot();
   });
 
-  it('updates createSsrFetch when added on tenant module', () => {
+  it('updates createSsrFetch when added on the module', () => {
```
Comment (JAdshead) [inline __tests__/server/utils/onModuleLoad.spec.jsx]:
```suggestion
  it('updates createSsrFetch when added on the root module', () => {
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 11]
```diff
@@ -294,7 +294,7 @@ describe('onModuleLoad', () => {
     expect(setCreateSsrFetch).toHaveBeenCalledWith(fakeCreateSsrFetch);
   });
 
-  it('sets CORS origins from tenant module', () => {
+  it('sets CORS origins from the module', () => {
```
Comment (JAdshead) [inline __tests__/server/utils/onModuleLoad.spec.jsx]:
```suggestion
  it('sets CORS origins from the root module', () => {
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.0/package.json [hunk 12]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "frank-lloyd-root",
   "version": "0.0.0",
-  "description": "Tenancy root module for integration tests",
+  "description": "Application root module for integration tests",
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.0/package.json]:
```suggestion
  "description": "Root module for integration tests",
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/components/FrankLloydRoot.jsx [hunk 13]
```diff
@@ -31,7 +31,7 @@ export function FrankLloydRoot({ children, config }) {
   return (
     <React.Fragment>
       <Helmet
-        title="Sample Tenancy"
+        title="Sample Application"
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/components/FrankLloydRoot.jsx]:
```suggestion
        title="Sample Root Module"
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.1/README.md [hunk 14]
```diff
@@ -1,3 +1,3 @@
 # frank-lloyd-root
 
-Tenancy root module for integration tests. This specific version has an invalid configuration.
\ No newline at end of file
+Application root module for integration tests. This specific version has an invalid configuration.
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.1/README.md]:
```suggestion
Root module for integration tests. This specific version has an invalid configuration.
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.1/package.json [hunk 15]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "frank-lloyd-root",
   "version": "0.0.1",
-  "description": "Tenancy root module for integration tests",
+  "description": "Application root module for integration tests",
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.1/package.json]:
```suggestion
  "description": "Root module for integration tests",
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.1/src/components/FrankLloydRoot.jsx [hunk 16]
```diff
@@ -26,7 +26,7 @@ export function FrankLloydRoot({ children, config }) {
   return (
     <React.Fragment>
       <Helmet
-        title="Sample Tenancy"
+        title="Sample Application"
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.1/src/components/FrankLloydRoot.jsx]:
```suggestion
        title="Sample Root Module"
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.2/README.md [hunk 17]
```diff
@@ -1,3 +1,3 @@
 # frank-lloyd-root
 
-Tenancy root module for integration tests. This specific version has a alternate valid configuration.
\ No newline at end of file
+Application root module for integration tests. This specific version has a alternate valid configuration.
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.2/README.md]:
```suggestion
Root module for integration tests. This specific version has a alternate valid configuration.
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.2/package.json [hunk 18]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "frank-lloyd-root",
   "version": "0.0.2",
-  "description": "Tenancy root module for integration tests",
+  "description": "Application root module for integration tests",
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.2/package.json]:
```suggestion
  "description": "Root module for integration tests",
```

File: prod-sample/sample-modules/frank-lloyd-root/0.0.2/src/components/FrankLloydRoot.jsx [hunk 19]
```diff
@@ -30,7 +30,7 @@ export function FrankLloydRoot({ children, config }) {
   return (
     <React.Fragment>
       <Helmet
-        title="Sample Tenancy"
+        title="Sample Application"
```
Comment (JAdshead) [inline prod-sample/sample-modules/frank-lloyd-root/0.0.2/src/components/FrankLloydRoot.jsx]:
```suggestion
        title="Sample Root Module"
```

File: src/server/middleware/sendHtml.js [hunk 20]
```diff
@@ -62,7 +62,7 @@ export function renderStaticErrorPage(res) {
     message = 'Sorry, we are unable to load this page at this time.';
   }
 
-  // TODO: allow tenant to provide custom error message and override default html
+  // TODO: allow application to provide custom error message and override default html
```
Comment (JAdshead) [inline src/server/middleware/sendHtml.js]:
```suggestion
  // TODO: allow root module to provide custom error message and override default html
```

File: src/server/utils/onModuleLoad.js [hunk 21]
```diff
@@ -67,7 +67,7 @@ export default function onModuleLoad({
 }) {
   const {
     [CONFIGURATION_KEY]: {
-      // Tenant Root Specific
+      // Application Root Specific
```
Comment (JAdshead) [inline src/server/utils/onModuleLoad.js]:
```suggestion
      // Root Module Specific
```

File: src/server/utils/safeRequest.js [hunk 22]
```diff
@@ -114,7 +114,7 @@ export const validateSafeRequestRestrictedAttributes = (requiredAttributes, modu
 
       if (!requestItemsIncluded) {
         missingRequestItemsMessages.push(
-          `Tenant root must extendSafeRequestRestrictedAttributes with ${key}: [${requiredItems}]`
+          `Application root must extendSafeRequestRestrictedAttributes with ${key}: [${requiredItems}]`
```
Comment (JAdshead) [inline src/server/utils/safeRequest.js]:
```suggestion
          `Root module must extendSafeRequestRestrictedAttributes with ${key}: [${requiredItems}]`
```

File: src/server/utils/stateConfig.js [hunk 23]
```diff
@@ -87,7 +87,7 @@ if (process.env.NODE_ENV === 'development' && fs.existsSync(pathToDevEndpoints))
 
 let stateConfigFromModule = { server: {}, client: {} };
 const configEnv = process.env.ONE_CONFIG_ENV;
-const makeConfigEnvError = () => new Error('Failed to parse an object in the tenant root configuration due to missing ONE_CONFIG_ENV.');
+const makeConfigEnvError = () => new Error('Failed to parse an object in the root configuration due to missing ONE_CONFIG_ENV.');
```
Comment (JAdshead) [inline src/server/utils/stateConfig.js]:
```suggestion
const makeConfigEnvError = () => new Error('Failed to parse an object in the root module configuration due to missing ONE_CONFIG_ENV.');
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 24]
```diff
@@ -327,7 +327,7 @@ describe('onModuleLoad', () => {
     expect(consoleInfoSpy).toHaveBeenCalledWith('Loaded module not-the-root-module@1.0.16');
   });
 
-  it('updates allowed safeRequest values from tenant root', () => {
+  it('updates allowed safeRequest values from the root', () => {
```
Comment (mtomcal) [inline __tests__/server/utils/onModuleLoad.spec.jsx]:
```suggestion
  it('updates allowed safeRequest values from the root module', () => {
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 25]
```diff
@@ -359,7 +359,7 @@ describe('onModuleLoad', () => {
     });
   });
 
-  it('sets configureRequestLog when given on tenant root', () => {
+  it('sets configureRequestLog when given on the root', () => {
```
Comment (mtomcal) [inline __tests__/server/utils/onModuleLoad.spec.jsx]:
```suggestion
  it('sets configureRequestLog when given on the root module', () => {
```

File: __tests__/universal/routes.spec.jsx [hunk 26]
```diff
@@ -44,7 +44,7 @@ describe('routes', () => {
 
   beforeEach(() => jest.clearAllMocks());
 
-  it('should set up the tenancy root route first', () => {
+  it('should set up the root route first', () => {
```
Comment (mtomcal) [inline __tests__/universal/routes.spec.jsx]:
```suggestion
  it('should set up the root module route first', () => {
```

File: __tests__/server/utils/onModuleLoad.spec.jsx [hunk 27]
```diff
@@ -185,7 +185,7 @@ describe('onModuleLoad', () => {
     expect(() => onModuleLoad({ module: { [CONFIGURATION_KEY]: configuration, [META_DATA_KEY]: { version: '1.0.7' } }, moduleName: 'my-awesome-module' })).not.toThrow();
   });
 
-  it('warns if a module that isn\'t the tenant root module attempts to provide externals', () => {
+  it('throws if a modules that isn\'t the root module attempts to provide externals', () => {
```
Comment (10xLaCroixDrinker) [inline __tests__/server/utils/onModuleLoad.spec.jsx:188]:
```suggestion
  it('warns if a module that isn\'t the root module attempts to provide externals', () => {
```

**Pull request merged**